02-library/cs506/kmeans.py

```diff
@@ -21,8 +24,16 @@ def update_centers(dataset, assignments):
     Compute the center for each of the assigned groups.
     Return `k` centers in a list
     """
-    raise NotImplementedError()
+    k = max(assignments) + 1
+    clusters = [[] for i in range(k)]
+    for i, index in enumerate(assignments):
```

Comment:
it's not very clear what the distinction is between `i` and `index`. Can you make the names more descriptive? For example:
```suggestion
    for pointIndex, pointAssignment in enumerate(assignments):
```